pkg/globalnet/controllers/node_controller.go

```diff
@@ -89,19 +90,25 @@ func (n *nodeController) process(from runtime.Object, numRequeues int, op syncer
 
 	// If the event corresponds to a different node which has globalIP annotation, release the globalIP back to Pool.
 	if node.Name != n.nodeName {
-		if existingGlobalIP := node.GetAnnotations()[GlobalIPKey]; existingGlobalIP != "" {
-			if op == syncer.Delete {
-				_ = n.pool.Release(existingGlobalIP)
-				return nil, false
-			} else {
-				_ = n.pool.Release(existingGlobalIP)
-				return n.updateNodeAnnotation(node, "")
-			}
+		if existingGlobalIP := node.GetAnnotations()[constants.SmGlobalIP]; existingGlobalIP != "" {
+			cniIfaceIP := node.GetAnnotations()[constants.CNIInterfaceIP]
+
+			requeue := n.flushRulesAndReleaseIPs(node.Name, numRequeues, func(allocatedIPs []string) error {
```

Comment:
We dont have to flush rules when the event corresponds to a different node as those rules would have been present on that OLD Gateway node and NOT on the current Active Gateway node. We just have to release the IP back to Pool and remove the annotation.

Reply:
ahh - right

Reply:
Removed the commit that changed it.